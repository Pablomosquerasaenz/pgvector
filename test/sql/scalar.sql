SET client_min_messages = warning;
CREATE EXTENSION IF NOT EXISTS vector;

SELECT '[1,2,3]'::vector + 10;
SELECT '[1,2,3]'::vector - 10;
SELECT '[1,2,3]'::vector * 10;
SELECT '[1,2,3]'::vector / 10;
SELECT 10 + '[1,2,3]'::vector;
SELECT 10 - '[1,2,3]'::vector;
SELECT 10 * '[1,2,3]'::vector;
SELECT 10 / '[1,2,3]'::vector;
