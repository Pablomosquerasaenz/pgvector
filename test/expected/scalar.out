SET client_min_messages = warning;
CREATE EXTENSION IF NOT EXISTS vector;
SELECT '[1,2,3]'::vector + 10;
  ?column?  
------------
 [11,12,13]
(1 row)

SELECT '[1,2,3]'::vector - 10;
  ?column?  
------------
 [-9,-8,-7]
(1 row)

SELECT '[1,2,3]'::vector * 10;
  ?column?  
------------
 [10,20,30]
(1 row)

SELECT '[1,2,3]'::vector / 10;
                           ?column?                            
---------------------------------------------------------------
 [0.10000000149011612,0.20000000298023224,0.30000001192092896]
(1 row)

SELECT 10 + '[1,2,3]'::vector;
  ?column?  
------------
 [11,12,13]
(1 row)

SELECT 10 - '[1,2,3]'::vector;
 ?column? 
----------
 [9,8,7]
(1 row)

SELECT 10 * '[1,2,3]'::vector;
  ?column?  
------------
 [10,20,30]
(1 row)

SELECT 10 / '[1,2,3]'::vector;
         ?column?          
---------------------------
 [10,5,3.3333332538604736]
(1 row)

